File: RemoteSignature- PDFBox/CSC-PDFBox/src/main/java/CSC_API_Utils.java
import JSON_FORMATS.CredentialsInfoResponse;
import com.itextpdf.io.codec.Base64;

import java.io.ByteArrayInputStream;
import java.io.FileInputStream;
import java.security.cert.*;

public class CSC_API_Utils {


 
    public static Certificate[] GetCertChain(String accessToken, String credentialId)
    {
        Certificate[] chain = new Certificate[] {GetSigningCertificate(accessToken,credentialId),GetCACertificate()};

        return chain;
    }



    public static Certificate GetSigningCertificate(String accessToken, String credentialId)
    {
        try {
            CredentialsInfoResponse info = CSC_API.CredentialsInfo(accessToken, credentialId);
            		//new CredentialsInfoResponse();

            byte []encodedCert = Base64.decode(info.cert.certificates[0]);
 
            ByteArrayInputStream inputStream =  new ByteArrayInputStream(encodedCert);
        
            CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
            X509Certificate cert = (X509Certificate)certificateFactory.generateCertificate(inputStream);

            return cert;
        }catch(Exception e)
        {
            e.printStackTrace();
            return null;
        }
    }
    public static Certificate GetCACertificate()
    {
        try {
        	CertificateFactory fact = CertificateFactory.getInstance("X.509");
        	FileInputStream is = new FileInputStream ("E:\\certificate.crt");
        	X509Certificate cert = (X509Certificate) fact.generateCertificate(is);
            return cert;
        }catch(Exception e)
        {
            e.printStackTrace();
            return null;
        }
    }

}

File: RemoteSignature iText7/RemoteSignatureApp/Classes/CSC_API_Utils.cs
using iText.Kernel.XMP.Impl;
using iText.Signatures;
using Org.BouncyCastle.Crypto;
using Org.BouncyCastle.Security;
using Org.BouncyCastle.X509;
using RemoteSignatureApp.Models;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Security.Cryptography.X509Certificates;
using System.Text;
using System.Threading.Tasks;

namespace RemoteSignatureApp.Classes
{
    class CSC_API_Utils
    {
        static readonly CSC_API_Client clt=new CSC_API_Client();
        public static async Task<Org.BouncyCastle.X509.X509Certificate[]> GetCertChainAsync(String accessToken, String credentialId)
        {
            Org.BouncyCastle.X509.X509Certificate[] chain = new Org.BouncyCastle.X509.X509Certificate[] {await GetSigningCertificateAsync(accessToken, credentialId) };

            return chain;
        }



        public static async Task<Org.BouncyCastle.X509.X509Certificate> GetSigningCertificateAsync(String accessToken, String credentialId)
        {
            try
            {
                Credentials_Info_Response info = await clt.Credentials_Info(accessToken, credentialId);


                byte[] bytes = Convert.FromBase64String(info.cert.certificates[0]);
               // var cert2 = new X509Certificate2(bytes);

                //cert.certificate[id certificatului selectat pt semnare)
                string encodedCert = Base64.Decode(info.cert.certificates[0]);

                Org.BouncyCastle.X509.X509Certificate cert = new X509CertificateParser().ReadCertificate(bytes);
                return cert;
            }
            catch (Exception e)
            {
                return null;
            }
        }
  

    }
}
